map/map.mbt

```diff
@@ -189,28 +201,30 @@ fn balance[K : Compare, V](
   fn single_r {
     k2, v2, Map::Tree(k1, v1, _, x, y), z => new(k1, v1, x, new(k2, v2, y, z))
     _, _, _, _ => abort("single_r error")
+
+    //      1                 2
+    //     / \              /   \
+    //    x   3            1     3
+    //       / \    -->   / \   / \
+    //      2   z        x  y1 y2  z
+    //     / \
+    //    y1 y2
   }
 
-  //      1                 2
```

Comment:
here is a format bug?

Reply:
It's a known bug in the formatter. Comments cannot be placed between local functions. It will be fixed soon.